src/main/java/calculator/CalculatorService.java

```diff
@@ -0,0 +1,132 @@
+package calculator;
+
+import com.google.code.mathparser.MathParser;
+import com.google.code.mathparser.MathParserFactory;
+import repository.Repository;
+import repository.RepositoryImpl;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static controllers.ControllerConstants.ABS_RANGE;
+import static controllers.ControllerConstants.DIVIDE;
+import static controllers.ControllerConstants.EMPTY_SYMBOL;
+import static controllers.ControllerConstants.EXPRESSION;
+import static controllers.ControllerConstants.MINUS;
+import static controllers.ControllerConstants.MULTIPLY;
+import static controllers.ControllerConstants.PLUS;
+import static controllers.ControllerConstants.STATUS_BAD_REQUEST;
+import static controllers.ControllerConstants.STATUS_CREATED;
+import static controllers.ControllerConstants.STATUS_FORBIDDEN;
+import static controllers.ControllerConstants.STATUS_OK;
+import static utils.ConversionUtil.deleteSpacesAndConvertListToString;
+
+
+public class CalculatorService implements Calculator {
+    private static final Repository REPOSITORY = RepositoryImpl.getInstance();
+
+    @Override
+    public int calculate(String ID) {
+        String expression = getFinalExpression(ID);
+        MathParser mathParser = MathParserFactory.create();
+
+        return mathParser.calculate(expression)
+                .doubleValue()
+                .intValue();
+    }
+
+    @Override
+    public int addData(String ID, String parameterName, String paramValue) {
+        String valueFromRepository = getValueFromRepository(ID, parameterName);
+        if (!isDataExists(ID)) {
+            REPOSITORY.putNewData(ID);
+        }
+        int statusCode = getStatusCode(valueFromRepository, parameterName, paramValue);
+        if (!isBadFormatExpression(paramValue, parameterName) && !isParameterHasOverLimitValue(paramValue)) {
+            REPOSITORY.update(ID, parameterName, paramValue);
+        }
+        return statusCode;
```

Comment:
I think it would be better to return Enum values { INSERTED, UPDATED } from service methods. And Controller should decide that status should be returned